### PR TITLE
ESQL: Copy blocks on expand

### PR DESCRIPTION
**Pull request by nik9000**
Until we get lovely reference tracking, let's copy blocks when "expanding" them. For now, that's simple and it works and helps make memory tracking cleaner. I expect we won't copy forever. We can likely share the array once we get reference tracking. But for now, let's copy!

Closes #100548

### CODE REVIEW

**Review by nik9000 — COMMENTED** [review]
@dnhatn, this is a rebase of #100548.

File: x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/MvExpandOperatorTests.java.orig [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+import java.util.Iterator;
+import java.util.List;
+
+import static org.elasticsearch.compute.data.BasicBlockTests.randomBlock;
+import static org.elasticsearch.compute.data.BasicBlockTests.valuesAtPositions;
+import static org.elasticsearch.compute.data.BlockTestUtils.deepCopyOf;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+
+public class MvExpandOperatorTests extends OperatorTestCase {
```
Comment (dnhatn) [inline x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/MvExpandOperatorTests.java.orig:28]: ++. Thanks for adding this test.

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks, Nik!

File: x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/MvExpandOperatorTests.java.orig (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/MvExpandOperatorTests.java.orig:28]: OoopS! That one was there before. thta's an `orig` from the patch. Ooops!

Comment (dnhatn) [reply x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/MvExpandOperatorTests.java.orig]: Ah

**Pull request merged**